fenzo-core/src/main/java/com/netflix/fenzo/plugins/WeightedAverageFitnessCalculator.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.fenzo.plugins;
+
+import java.util.List;
+
+import com.netflix.fenzo.TaskRequest;
+import com.netflix.fenzo.TaskTrackerState;
+import com.netflix.fenzo.VMTaskFitnessCalculator;
+import com.netflix.fenzo.VirtualMachineCurrentState;
+
+/**
+ * A fitness calculator that calculates the weighted average of multiple fitness calculators.
+ */
+public class WeightedAverageFitnessCalculator implements VMTaskFitnessCalculator {
+
+    private final List<WeightedFitnessCalculator> calculators;
+
+    public WeightedAverageFitnessCalculator(List<WeightedFitnessCalculator> calculators) {
+        this.calculators = calculators;
+        if (calculators == null || calculators.isEmpty()) {
+            throw new IllegalArgumentException("There must be at least 1 calculator");
+        }
+        double sum = calculators.stream()
+                .map(WeightedFitnessCalculator::getWeight)
+                .mapToDouble(Double::doubleValue).sum();
+        if (sum != 1.0) {
```

Comment:
I don't see a reason why we can't normalize the sum and not require the sum to be 1.0. For example, if weights for 3 calculators were 2, 3, and 6, we can normalize it by dividing things by 11. Isn't it?

Reply:
I am validating to make sure the user is normalizing their weights to 1.0 to prevent mistakes.